WIP: Order results by search field

rossjones opened the pull request.

rossjones committed:
Order results by search field

simonwo commented on prototypes/v2/src/controllers.ts:
```diff
@@ -55,8 +56,7 @@ export module Controllers {
 
-      let query =
-        "select rank, PRODUCT_ID from search where search match ? order by rank LIMIT ? OFFSET ?;";
+      let query = `select rank, PRODUCT_ID from search where search match ? order by ${sortBy} LIMIT ? OFFSET ?;`;
```
Isn't this an injection issue?
```
?sortBy=rank%3BDROP%20TABLE%20products%3B--

rossjones replied:
Yep, absolutely. I've shelved this as a WIP while I rethink it a bit.

rossjones committed:
Remove authentication on base prototype (index) (#24)

alaric-rd committed:
GMDN, GUDID and MHRA importers (#21)

* GMDN and GUDID importers

* .gitignore files to avoid committing sqlite databases by accident

* Additional GUDID files imported

* Import scripts for the MHRA data snapshot we have so far

* README for data importers

* Added comment about the two tables we've ignored thus far

---------

Co-authored-by: Alaric Snell-Pym <alaric@snell-pym.org.uk>

Weeza committed:
Update product.mustache (#25)

* rearranged and separated sections
* removed downloads
* minor content tweaks

mikieet6 committed:
Updates to filters (#26)

mikieet6 committed:
Initial updates for review (#27)

mikieet6 committed:
Fixing homepage and redundant index pages on v1 and v2 (#28)

* Initial updates for review

* Moving conditional field

* Fixing links on basic homepage and skipping v1 and v2 index pages

rossjones committed:
Add excel and CSV download (#29)

This is used for search results (and single product detail page).
Currently the number of rows is capped at 5000 until we can find a way
to lazily load the data, and in such a way we don't end up with files
that are hundreds of megabytes.

rossjones committed:
Provide a map to lookup search column

Rather than allow the front-end to specify which search column to sort
by, provide a map.

Available options are

* manufacturer - manufacturer name
* term - the GMDN term
* product - the product's brand name

rossjones commented:
Closing in favour of a safer one
